module/zfs/arc.c

```diff
@@ -7305,8 +7305,14 @@ arc_init(void)
 	arc_lowmem_init();
 #endif
 
-	/* Set min cache to 1/32 of all memory, or 32MB, whichever is more. */
-	arc_c_min = MAX(allmem / 32, 2ULL << SPA_MAXBLOCKSHIFT);
+	/*
+	 * Set min cache to 1/8 of all memory, or 32MB, whichever is more.
+	 * Note that this needs to be large enough to contain the entire dbuf
+	 * cache, plus the ARC's copy of this data.  By default the dbuf
+	 * cache is 3/64ths of memory (see dbuf_cache_shift), so this should
+	 * be at least 6/64ths of memory.
+	 */
+	arc_c_min = MAX(allmem / 8, 2ULL << SPA_MAXBLOCKSHIFT);
```

Comment:
shouldn't `arc_tuning_update()` include a rule check to make sure `arc_c_min` is large enough to handle the dbuf cache?

Reply:
That's a good question.  It currently allows arc_c_min to be set as low as 32MB, even if that's below the old default of 1/32nd of RAM.  So I'm preserving the flexibility of the current behavior.  But I'd also be open to input on how to define what's "reasonable" for this value.  It seems like the value we've been using for quite some time (1/32nd of RAM) is in practice "reasonable enough" for a lot of workloads, since we haven't noticed this before now?